core/backend/embeddings.go

```diff
@@ -3,13 +3,13 @@ package backend
 import (
 	"fmt"
 
-	config "github.com/go-skynet/LocalAI/core/config"
-	"github.com/go-skynet/LocalAI/core/options"
+	"github.com/go-skynet/LocalAI/core/config"
+
 	"github.com/go-skynet/LocalAI/pkg/grpc"
 	model "github.com/go-skynet/LocalAI/pkg/model"
 )
 
-func ModelEmbedding(s string, tokens []int, loader *model.ModelLoader, c config.Config, o *options.Option) (func() ([]float32, error), error) {
+func ModelEmbedding(s string, tokens []int, loader *model.ModelLoader, c config.BackendConfig, o *config.ApplicationConfig) (func() ([]float32, error), error) {
```

Comment:
I don't want to sound too much picky here, but if ApplicationConfig here are set by the code only (and not read by any external config file) then it makes sense to keep under "options". So we can distinguish what's coming from a configuration file (e.g. a YAML file) and what is instead an option that is set when calling constructors in the code

Reply:
I'll give this some thought today - there _is_ a plan to set ApplicationConfig variables from two different places. Primarily, these will still come from the command line arguments and their defaults, but I've introduced the `configuration` directory and `startup/config_file_watcher` (which, honestly, may belong under the `config` package now...) to handle runtime modifications to the non-backend-specific ApplicationConfig. Currently this is limited to enabling/disabling external backends and api_keys, but I do intend to expand this over time - for example, I currently am planning to have a file in this directory control what queues are active for MQTT

Reply:
Watcher seems to cause circular reference issues if moved to config, leaving it in startup for this PR.